tedge/src/certificate.rs

```diff
@@ -333,55 +335,54 @@ impl CertError {
 }
 
 impl BuildCommand for TEdgeCertOpt {
-    fn build_command(self, config: TEdgeConfig) -> Result<Box<dyn Command>, ConfigError> {
-        let cmd =
-            match self {
-                TEdgeCertOpt::Create { id } => {
-                    let cmd = CreateCertCmd {
-                        id,
-                        cert_path: config.device.cert_path.ok_or_else(|| {
-                            ConfigError::ConfigNotSet {
-                                key: String::from(DEVICE_CERT_PATH),
-                            }
-                        })?,
-                        key_path: config.device.key_path.ok_or_else(|| {
-                            ConfigError::ConfigNotSet {
-                                key: String::from(DEVICE_KEY_PATH),
-                            }
-                        })?,
-                    };
-                    cmd.into_boxed()
-                }
+    fn build_command(self, context: BuildCommandContext) -> Result<Box<dyn Command>, ConfigError> {
+        let cmd = match self {
+            TEdgeCertOpt::Create { id } => {
+                let cmd = CreateCertCmd {
+                    id,
+                    cert_path: context.config.device.cert_path.ok_or_else(|| {
```

Comment:
Why a direct access to the config dto struct `config.device.cert_path` as before?

* I was expecting here a call to `config.query(DeviceCertPathSetting)?`.
* Is this PR just an intermediate step.

Reply:
This is not handled by this PR.